doc/reference/foreign-archives.rst

```diff
@@ -0,0 +1,49 @@
+Foreign Archives
+----------------
+
+You can also specify archives of separately compiled foreign object files
```

Comment:
"also" is weird here, as there is nothing in this text to relate to now that it was split. Maybe a short introduction what foreign archives are and what they can be used for would be a better motivation paragraph.